### PR TITLE
Up solvercore patch

### PR DESCRIPTION
**Pull request by tmigot**

### CODE REVIEW

**Comment by tmigot** [comment]
@JSOBot runtests

File: src/api.jl [hunk 1]
```diff
@@ -106,15 +106,20 @@ function knitro(nls::AbstractNLSModel; kwargs...)
...
-    set_multipliers!(
-      stats,
-      fstats.multipliers[(nls.nls_meta.nequ + 1):end],
-      has_bounds(nls) ? fstats.multipliers_L[1:(nls.meta.nvar)] : similar(fstats.multipliers_L),
-      has_bounds(nls) ? fstats.multipliers_U[1:(nls.meta.nvar)] : similar(fstats.multipliers_U),
-    )
+    set_multipliers!(stats, fstats.multipliers[(nls.nls_meta.nequ + 1):end])
+    if has_bounds(nls)
+      fstats.bounds_multipliers_reliable || error("bounds multipliers unreliable")
+      set_multipliers!(
```
Comment (dpo) [inline src/api.jl:117]: set_bounds_multipliers ?

Comment (dpo) [inline src/api.jl:117]:
```suggestion
      set_bounds_multipliers!(
```

**Comment by dpo** [comment]
@JSOBot runtests

File: src/api.jl [hunk 2]
```diff
@@ -106,15 +106,20 @@ function knitro(nls::AbstractNLSModel; kwargs...)
...
+    fstats.dual_residual_reliable || error("dual residual unreliable")
+    set_dual_residual!(stats, fstats.dual_feas)
     fstats.multipliers_reliable || error("multipliers unreliable")
-    set_multipliers!(
-      stats,
-      fstats.multipliers[(nls.nls_meta.nequ + 1):end],
-      has_bounds(nls) ? fstats.multipliers_L[1:(nls.meta.nvar)] : similar(fstats.multipliers_L),
-      has_bounds(nls) ? fstats.multipliers_U[1:(nls.meta.nvar)] : similar(fstats.multipliers_U),
-    )
+    set_multipliers!(stats, fstats.multipliers[(nls.nls_meta.nequ + 1):end])
```
Comment (dpo) [inline src/api.jl:114]:
```suggestion
    set_constraint_multipliers!(stats, fstats.multipliers[(nls.nls_meta.nequ + 1):end])
```

**Comment by dpo** [comment]
@JSOBot runtests

**Comment by dpo** [comment]
Fixing missing export in https://github.com/JuliaSmoothOptimizers/SolverCore.jl/pull/44

**Comment by dpo** [comment]
@JSOBot runtests

**Comment by dpo** [comment]
@JSOBot runtests

**Pull request merged**